src/librustdoc/html/render/mod.rs

```diff
@@ -2032,7 +2038,15 @@ fn sidebar_deref_methods(cx: &Context<'_>, out: &mut Buffer, impl_: &Impl, v: &V
                     .filter(|i| i.inner_impl().trait_.is_some())
                     .find(|i| i.inner_impl().trait_.def_id_full(c) == c.deref_trait_did)
                 {
-                    sidebar_deref_methods(cx, out, target_deref_impl, target_impls);
+                    if already_seen.insert(target_did.clone()) {
+                        sidebar_deref_methods(
+                            cx,
+                            out,
+                            target_deref_impl,
+                            target_impls,
+                            already_seen,
+                        );
+                    }
                 }
             }
```

Comment:
Please instead just remove this whole block, rustdoc no longer renders recursive Deref after https://github.com/rust-lang/rust/pull/84867 and removing the block should fix https://github.com/rust-lang/rust/issues/85037. Please add a test for that issue too.